flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -19,33 +19,35 @@
 package org.apache.flink.table.catalog;
 
 import org.apache.flink.annotation.Internal;
-import org.apache.flink.table.api.CatalogAlreadyExistsException;
 import org.apache.flink.table.api.CatalogNotExistException;
-import org.apache.flink.table.api.ExternalCatalogAlreadyExistException;
 import org.apache.flink.table.api.TableSchema;
 import org.apache.flink.table.catalog.exceptions.CatalogException;
-import org.apache.flink.table.catalog.exceptions.DatabaseNotExistException;
 import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+import org.apache.flink.table.factories.TableFactoryUtil;
 import org.apache.flink.table.operations.CatalogTableOperation;
+import org.apache.flink.table.sinks.TableSink;
 import org.apache.flink.util.StringUtils;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
+import static java.lang.String.format;
```

Comment:
can we revert the newly added static import besides Preconditions? I think we should avoid static import as much as possible as they make the code not noticeably shorter but less readable and maintainable.

Reply:
I tend to disagree. My experience is, that it is usually the other way round(with some rare exceptions). Static imports make the code more readable and maintainable.

Reply:
I believe we should only allow two cases to use static import: a) static constants b) extraordinarily common static helpers that don't cause confusions at all for any java developers in some specific context, like assertXxxx in junit tests, and checkArgument/checkNotNull for validating input args.

When I read this code, it indeed confused me that I had to look around for where the `format()` and `asList()` come from.

Reply:
I think we should apply common sense here. In my experience it makes it much cleaner when we statically import well known methods or constants (like e.g. `String.format()`/`Assert.assertTrue`\`Collectors.toList()` etc.). There is very little chance anyone will confuse 
```
CatalogException(format(
	"A database with name [%s] does not exist in the catalog: [%s].",
	databaseName,
	currentCatalogName))
```
with any other `format`.

Reply:
I rephrased my previous comment and I think we agree in the principal. I guess the only difference is if we assume `String#format` and `Arrays#asList` to be well known by majority of java developers.

They really broadly used in Flink's code base(and not only in Flink's) and I truly believe they are rarely confused.

Reply:
Anyway, after a quick scan to be remain consistent with other places, I will change it.